src/main/java/org/spongepowered/api/text/dictionary/Dictionary.java

```diff
@@ -0,0 +1,394 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.text.dictionary;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.Text.Placeholder;
+import org.spongepowered.api.text.Texts;
+import org.spongepowered.api.text.Texts.FormatBuilder;
+import org.spongepowered.api.text.format.TextColors;
+import org.spongepowered.api.util.BiFunctions;
+import org.spongepowered.api.util.Functions;
+import org.spongepowered.api.util.command.CommandSource;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * Utility class to allow plugin developers to easily add multi language support
+ * to their plugins.
+ */
+public class Dictionary implements Function<String, Function<Locale, Text>> {
```

Comment:
This entire class is an optional addition and may be removed from this PR if its not considered useful.
(It could be moved to the example code section in that case)
Its more or less a showcase how to achieve multilanguage support. See my test plugin/PR description for its use/usage. (And it will probably be the class/implementation I willl use for my own plugins.)